torchrl/modules/models/multiagent.py

```diff
@@ -230,6 +230,221 @@ def forward(self, *inputs: Tuple[torch.Tensor]) -> torch.Tensor:
+
+    Args:
+        n_agents (int): number of agents.
+        centralised (bool): If ``True``, each agent will use the inputs of all agents to compute its output, resulting in input of shape ``(*B, n_agents * channels, x, y)``. Otherwise, each agent will only use its data as input.
+        share_params (bool): If ``True``, the same :class:`~torchrl.modules.ConvNet` will be used to make the forward pass
+            for all agents (homogeneous policies). Otherwise, each agent will use a different :class:`~torchrl.modules.ConvNet` to process
+            its input (heterogeneous policies).
+        device (str or torch.device, optional): device to create the module on.
+        num_cells (int or Sequence[int], optional): number of cells of every layer in between the input and output. If
+            an integer is provided, every layer will have the same number of cells. If an iterable is provided,
+            the linear layers ``out_features`` will match the content of ``num_cells``.
+        kernel_sizes (int, Sequence[Union[int, Sequence[int]]]): Kernel size(s) of the convolutional network.
+            Defaults to ``5``.
+        strides (int or Sequence[int]): Stride(s) of the convolutional network. If iterable, the length must match the
+            depth, defined by the num_cells or depth arguments.
+            Defaults to ``2``.
+        activation_class (Type[nn.Module]): activation class to be used.
+            Default to :class:`torch.nn.ELU`.
+        **kwargs: for :class:`~torchrl.modules.models.ConvNet` can be passed to customize the ConvNet.
+
+
+    Examples:
+        >>> import torch
+        ... from torchrl.modules import MultiAgentConvNet
+        ... batch = (3,2)
+        ... n_agents = 7
+        ... channels, x, y = 3, 100, 100
+        ... obs = torch.randn(*batch, n_agents, channels, x, y)
+        >>> # First lets consider a centralised network with shared parameters.
+        >>> cnn = MultiAgentConvNet(
+        ...     n_agents,
+        ...     centralised = True,
+        ...     share_params = True
+        ... )
+        ... print(cnn)
+        MultiAgentConvNet(
+            (agent_networks): ModuleList(
+                (0): ConvNet(
+                (0): LazyConv2d(0, 32, kernel_size=(5, 5), stride=(2, 2))
+                (1): ELU(alpha=1.0)
+                (2): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (3): ELU(alpha=1.0)
+                (4): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (5): ELU(alpha=1.0)
+                (6): SquashDims()
+                )
+            )
+        )
+        >>> result = cnn(obs)
+        ... print(result.shape)
+        torch.Size([3, 2, 7, 2592])
+        >>> # where the final dimension is determined by the input x
+        >>> print(all(result[0,0,0] == result[0,0,1]))
+        True
+        >>> # Since both shared observations and parameters leads to all agents having identical outputs (eg. for a value function)
```

Comment:
same here
the flow seems a bit broken

If we go for comments in code, it should flow like comments in a script. Otherwise we do
```

    Some story about our features.
    Examples:
        >>> # a module can get smth as input
        >>> module(smth)

    Some other story about other features.
    Examples:
        >>> # we can reuse module here:
        >>> assert module.attribute is "awesome"
```

hope that helps

Reply:
Clarified comments and adjusted order.